File: eAirFlow.Model/Models/Luggage.cs
using System;
using System.Collections.Generic;

namespace eAirFlow.Model.Models
{

    public class Luggage
    {
        public int LuggageId { get; set; }

        public int? UserId { get; set; }

        public string? Description { get; set; }

        public string? ImageUrl { get; set; }

        public string? StateMachine { get; set; }

        public int? AirportId { get; set; }
        public Airport? Airport { get; set; }
        public User? User { get; set; }



    }
}

File: UI/eairflow_mobile/lib/models/luggage.dart
import 'package:eairflow_mobile/models/airport.dart';
import 'package:eairflow_mobile/models/user.dart';

class Luggage {
  int? luggageId;
  String? description;
  String? status;
  int? airportId;
  int? userId;
  Airport? airport;
  User? user;
  String? stateMachine;
  String? imageUrl;

  Luggage({
    this.luggageId,
    this.description,
    this.status,
    this.airportId,
    this.userId,
    this.airport,
    this.user,
    this.stateMachine,
    this.imageUrl,
  });

  factory Luggage.fromJson(Map<String, dynamic> json) => Luggage(
        luggageId: json['luggageId'],
        description: json['description'],
        status: json['status'],
        airportId: json['airportId'],
        userId: json['userId'],
        airport: json['airport'] != null
            ? Airport.fromJson(json['airport'])
            : null,
        user: json['user'] != null
            ? User.fromJson(json['user'])
            : null,
        stateMachine: json['stateMachine'],
        imageUrl: json['imageUrl'] as String?,
      );
}
